### DISCUSSION TITLE
NavigationStack inside sheet appends first destination without transition on iOS 16

### DISCUSSION THREAD

**Discussion by KirillZholnerovich** [pointfreeco/swift-composable-architecture#3232] [General]
### Description

It works well on iOS 17,  but does not work correctly on iOS16

I've created an example project to reproduce this bug

Screen A
`
@Reducer
public struct ScreenAFeature {
	@ObservableState
	public struct State: Equatable {
		@Presents public var screenB: ScreenBFeature.State? = nil
		public init() {}
	}
	
	public enum Action {
		case screenB(PresentationAction<ScreenBFeature.Action>)
		case showButtonTapped
	}
		
	public init() {}
	
	public var body: some ReducerOf<Self> {
		Reduce { state, action in
			switch action {
			case .screenB:
				return .none
				
			case .showButtonTapped:
				state.screenB = .init()
				return .none
			}
		}
		.ifLet(\.$screenB, action: \.screenB) {
			ScreenBFeature()
		}
	}
}

public struct ScreenAView: View {
	@Perception.Bindable var store: StoreOf<ScreenAFeature>
	
	public init(store: StoreOf<ScreenAFeature>) {
		self.store = store
	}
	
	public var body: some View {
		WithPerceptionTracking {
			VStack {
				Text("Screen A")
				Button("Show B") {
					self.store.send(.showButtonTapped)
				}
			}
			.sheet(
				item: self.$store.scope(
					state: \.screenB,
					action: \.screenB
				)
			) { store in
				ScreenBView(store: store)
			}
		}
	}
}
`
Screen B
`
@Reducer
public struct ScreenBFeature {
	@ObservableState
	public struct State: Equatable {
		public var path: StackState<Path.State> = StackState()
		
		public init() {}
	}
	
	public enum Action {
		case path(StackAction<Path.State, Path.Action>)
		case toCButtonTapped
	}
	
	@Reducer(state: .equatable)
	public enum Path {
		case screenC(ScreenCFeature)
	}
	
	public init() {}
	
	public var body: some ReducerOf<Self> {
		Reduce { state, action in
			switch action {
			case .path:
				return .none
				
			case .toCButtonTapped:
				state.path.append(.screenC(.init()))
				return .none
			}
		}
		.forEach(\.path, action: \.path)
	}
}

public struct ScreenBView: View {
	@Perception.Bindable public var store: StoreOf<ScreenBFeature>
		
	public init(store: StoreOf<ScreenBFeature>) {
		self.store = store
	}
	
	public var body: some View {
		WithPerceptionTracking {
			NavigationStack(path: self.$store.scope(state: \.path, action: \.path)) {
				VStack {
					Text("Screen B")
										
					NavigationLink(state: ScreenBFeature.Path.State.screenC(.init())) {
						Text("To Screen C")
					}
				}
			} destination: { store in
				WithPerceptionTracking {
					switch store.case {
					case .screenC(let store):
						ScreenCView(store: store)
					}
				}
			}
		}
	}
}
`

Screen C
`
@Reducer
public struct ScreenCFeature {
	@ObservableState
	public struct State: Equatable {
		public init() {}
	}
	
	public enum Action {}
	
	public init() {}
	
	public var body: some ReducerOf<Self> {
		Reduce { state, action in
			switch action {
			default: return .none
			}
		}
	}
}

public struct ScreenCView: View {
	public let store: StoreOf<ScreenCFeature>
	
	public init(store: StoreOf<ScreenCFeature>) {
		self.store = store
	}
	
	public var body: some View {
		WithPerceptionTracking {
			Text("Screen C")
		}
	}
}
`

Behaviour on iOS 16

https://github.com/pointfreeco/swift-composable-architecture/assets/14135206/52e151c1-515c-422e-a4dc-7ad171f32c75

Behaviour on iOS 17

https://github.com/pointfreeco/swift-composable-architecture/assets/14135206/30ec6a8d-44c4-4f3a-85eb-c48651cc58c3

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Appearing works with transition.

### Actual behavior

Appearing works without transition.

### Steps to reproduce

1. Tap "Show B" button on ScreenAView.
2. Tap "To Screen C" button on ScreenBView.

### The Composable Architecture version information

1.11.2

### Destination operating system

iOS 16

### Xcode version information

15.4

### Swift Compiler version information

```shell
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @KirillZholnerovich, this is a bug in vanilla SwiftUI and there’s nothing we can do to fix it in the library. 

I am going to convert this to a discussion since it’s not an issue with the library. Feel free to continue the conversation over there.
